### DISCUSSION TITLE
fetch in useEffect ?

### DISCUSSION THREAD

**Discussion by grallc** [vercel/next.js#19424] [Help] [answered]
Hello!
I've been playing with NextJS today, with the idea to move from CRA to Next, and to be honest I really like it so far!

<del> However, I'm struggling with something : I need to fetch data whenever my app loads by making an API request. I've therefore looked into the `getInitialProps`, since it's not possible to use the `getServerSideProps` in _app.
I `think` I need to put my API request in my _app file since I need to fetch the data whichever the first page the user arrives in.</del>

<del>The problem is that I need to get the user's request to get the subdomain my user arrived through. The `req` object is available on `getServerSideProps`, but doesn't seem to be available on `getInitialProps`.</del>

<del>How can I achieve what I want (cf. getting user's subdomain) ? Am I doing something wrong ? </del>
Solved by https://github.com/vercel/next.js/discussions/10874#discussioncomment-83099 :)

Other question, which is not really related to my problem : is it a "bad" habit to fetch data using `fetch` in an `useEffect` if this data doesn't need to be server-side rendered  ?

Thanks!

**Comment by rafaelalmeidatk** [comment, marked as answer]
> is it a "bad" habit to fetch data using fetch in an useEffect if this data doesn't need to be server-side rendered ?

If the data don't need to be SSR then it is better to fetch in an `useEffect` since the users can start to see something being rendered before these requests are complete, you will often see these requests being done with some libraries like `swr` or `react-query`

**Reply by leerob** [reply]
You can also statically generate your site using `getStaticProps`/`getStaticPaths`. 

https://nextjs.org/docs/basic-features/data-fetching
